pkg/controller/manager/setup.go

```diff
@@ -1,12 +1,25 @@
 package manager
 
 import (
+	"github.com/urfave/cli"
+
 	"github.com/harvester/harvester-network-controller/pkg/config"
 	"github.com/harvester/harvester-network-controller/pkg/controller/manager/clusternetwork"
 	"github.com/harvester/harvester-network-controller/pkg/controller/manager/node"
+	"github.com/harvester/harvester-network-controller/pkg/controller/manager/vip"
 )
 
-var RegisterFuncList = []config.RegisterFunc{
-	clusternetwork.Register,
-	node.Register,
+var registerMap = map[string]config.RegisterFunc{
```

Comment:
I think we don't need to change RegisterFuncList to a map. We can append `vip.Register` to the funcList when `--enable-vip-controller` is set